core/state/database.go

```diff
@@ -0,0 +1,41 @@
+package state
+
+import (
+	"github.com/NethermindEth/juno/core/crypto"
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/NethermindEth/juno/core/trie2"
+	"github.com/NethermindEth/juno/core/trie2/triedb"
+	"github.com/NethermindEth/juno/core/trie2/trieutils"
+	"github.com/NethermindEth/juno/db"
+)
+
+const (
+	ClassTrieHeight           = 251
+	ContractTrieHeight        = 251
+	ContractStorageTrieHeight = 251
+)
+
+type StateDB struct {
+	disk       db.KeyValueStore
+	triedb     *triedb.Database
+	stateCache *stateCache
+}
+
+func NewStateDB(disk db.KeyValueStore, triedb *triedb.Database) *StateDB {
+	return &StateDB{disk: disk, triedb: triedb, stateCache: newStateCache()}
+}
```

Comment:
`StateDB` is meant to be a long persistent object used across runtimes? If true, then I think we should return by reference 👍 like right now